clients/src/main/java/org/apache/kafka/common/protocol/Protocol.java

```diff
@@ -1141,6 +1199,17 @@
 
     public static final Schema UPDATE_METADATA_REQUEST_PARTITION_STATE_V3 = UPDATE_METADATA_REQUEST_PARTITION_STATE_V2;
 
+    public static final Schema UPDATE_METADATA_REQUEST_PARTITION_STATE_V4 =
```

Comment:
Sure. I added the following comment for `UPDATE_METADATA_REQUEST_PARTITION_STATE_V4`, `UPDATE_METADATA_REQUEST_V4` and `METADATA_RESPONSE_V5`.

`... added a per-partition offline_replicas field. This field specifies the list of replicas that are offline`